CRM/Contact/BAO/ContactType.php

```diff
@@ -886,11 +869,37 @@ public static function deleteCustomRowsOfSubtype($gID, $subtypes = [], $subtypes
    *   Entity id.
    *
    * @return null|string
+   *
+   * @throws \CRM_Core_Exception
    */
   public static function deleteCustomRowsForEntityID($customTable, $entityID) {
     $customTable = CRM_Utils_Type::escape($customTable, 'String');
     $query = "DELETE FROM {$customTable} WHERE entity_id = %1";
     return CRM_Core_DAO::singleValueQuery($query, [1 => [$entityID, 'Integer']]);
   }
 
+  /**
+   * Get all contact types, leveraging caching.
+   *
+   * @return array
+   *
+   * @throws \API_Exception
+   */
+  protected static function getAllContactTypes() {
+    if (!Civi::cache('contactTypes')->has('all')) {
```

Comment:
@eileenmcnaughton do we need to bust this cache whenever a contact type is enabled or disabled or deleted or does the system already do that sensibly?

Reply:
It's already doing ```    Civi::cache('contactTypes')->clear();```